### PR TITLE
DApp Staking v2 - Top header page

### PR DESCRIPTION
**Pull request by sirius651**
**Pull Request Summary**

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

스크린샷 2022-09-13 오후 2 57 30

스크린샷 2022-09-13 오후 2 57 43

스크린샷 2022-09-13 오후 2 58 23

### CODE REVIEW

**Comment by sirius651** [comment]
I'm still continuing binding props for the components & separate common component into library later. Just wanted to get review for look & feel things

**Comment by Kahonnohak** [comment]
> I'm still continuing binding props for the components & separate common component into library later. Just wanted to get review for look & feel things

I like it a lot! Great work. There are a few points mostly responsive movements that need more attention. Should I start some feedback or are you in the middle of something that I should wait?

**Comment by Kahonnohak** [comment]
Hi Thanks @sirius651 Great work!

I have a few requests but nothing serious.

I would like to change the content width which I highlighted in the sheet. i'm sure it is possible but let me know if anything is not cleared. Let's sync tomorrow morning to avoid misunderstanding. 

https://docs.google.com/spreadsheets/d/11kr8g-ZJmH9Oon-R2iM75wJiBoP81l0tVbyAZaQw6Hk/edit#gid=630184198

**Comment by sirius651** [comment]
> I have a few requests but nothing serious.
> 
> I would like to change the content width which I highlighted in the sheet. i'm sure it is possible but let me know if anything is not cleared. Let's sync tomorrow morning to avoid misunderstanding.
> 
> https://docs.google.com/spreadsheets/d/11kr8g-ZJmH9Oon-R2iM75wJiBoP81l0tVbyAZaQw6Hk/edit#gid=630184198

Thanks for summarizing. I covered everything except 5, which is not sure what for. and 4 is in ui lib. will apply later. please check again

**Comment by Kahonnohak** [comment]
Thank you! 

There are small things to point out but let's move to the others for now.
They are between desktop and mobile so not many people would be seeing these. Can we leave the PR as it is for now? 

Screenshot 2022-09-16 at 10 51 45

Screenshot 2022-09-16 at 10 51 26

For the tab, it is a bit odd to see the texts wobbling when changing the font weight. This is not also critical but I would like to fix at some point.

https://gyazo.com/81b406bf327d4d3dd0d4e5a0434261c9

**Comment by sirius651** [comment]
스크린샷 2022-09-23 오후 9 03 36
스크린샷 2022-09-23 오후 9 03 02
스크린샷 2022-09-23 오후 9 03 22
ent.com/1606820/191969778-7b480af3-01d5-4b9c-a2e4-c6996e2eec47.png)

@Kahonnohak  I think top page UI implementation is finished, can you review on it?

**Comment by sirius651** [comment]
@Kahonnohak I think it's ready for the check on all your feedback except discussion A, which should more thinking

**Comment by Kahonnohak** [comment]
@sirius651 Thank you!

The carousel has a problem. We don't want the scroller. 

Screenshot 2022-09-28 at 09 46 03
Screenshot 2022-09-28 at 09 46 09

I got this sample please take a look. We son't need the dots at the bottom, we don't need the navigation arrow icons. 
https://gyazo.com/c1d63f6c104663e511be4f276c00f111

Ref:https://codepen.io/niteshgourofficial/pen/KeyEJv

**Comment by Kahonnohak** [comment]
Lovely! I got a few things (very details)

1) Hero images
Now I got Astar and Shiden heroes.
See below - could you place the image at the very top under the header? So I am hoping to see background blur here. 
- I have changed the height as I was not planning to have the header at the beginning. 
- The catch copy will be replace and possibly animation if time allows
Screenshot 2022-09-28 at 22 08 26

Please find the images here
https://drive.google.com/drive/folders/1mjgPruPuIjmF6H9XzaYi7RQJfUUE-fQd

2) Ads
This will be provided along with other thumbnails

3)"My Staking" also needs the header bar.

4)Padding
Could you make sure all paddings between contents are 80px (desktop and mobile)

5) Card
- It's already beautiful but if possible could you try to bring the dApp icon in centre of the height please?
- Could you place 4 cards when Max width(1296 in my calculation) . And start responsive with three columns?
- After breaks please make t
Screenshot 2022-09-28 at 21 56 57
he padding between the card to 16px from 24px

6) Thumbnail +link
Could you use the same rule as the card, please? 
- Responsive (it is)
- After breaks, the minimum size is the same as the card and padding to 16px
- Scroll
Screenshot 2022-09-28 at 22 04 04

**Comment by Kahonnohak** [comment]
The morning I'm adding some animation requests here.  Not all are must-have, they are nice -to-have. But we should try a little bit for the top page. your suggestions are also very welcome!

1) We need Skelton 

2)After loading
3)When scrolling the next content will fade in nicely simply gently. 
https://codepen.io/ugokuweb/pen/abpZpzj
Japanese but it has two animations (2 and3) in it.

4) Title animation
https://gyazo.com/b86061e5e4e3889a9e23d33b04b6c364

**Comment by sirius651** [comment]
> * https://gyazo.com/b86061e5e4e3889a9e23d33b04b6c364

I think that's more cool. currently implement that animation as you refer.

**Comment by sirius651** [comment]
regarding the issue `Deploy to Firebase Hosting on PR / build_and_preview (pull_request)` is failing, could you check? @niklabh

**Comment by niklabh** [comment]
fixed

File: src/components/dapp-staking-v2/my-staking/MyDapps.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <div>
+    <template v-if="width >= screenSize.lg">
+      <div class="table-container">
+        <table id="my-table">
+          <thead>
+            <tr>
+              <th>dApps</th>
```
Comment (impelcrypto) [inline src/components/dapp-staking-v2/my-staking/MyDapps.vue:8]: Let's display texts from i18 file

File: src/components/dapp-staking-v2/my-staking/MyDapps.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              <th>Staked Amount</th>
+              <th>Total Earned</th>
+              <th>Manage</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr v-for="t in items" :key="t.id">
+              <td>{{ t.name }}</td>
+              <td>{{ t.stakedAmount.toLocaleString() }} ASTR</td>
```
Comment (impelcrypto) [inline src/components/dapp-staking-v2/my-staking/MyDapps.vue:17]: You can change like [this](https://github.com/AstarNetwork/astar-apps/blob/feat/dapp-staking-v2/src/components/assets/NativeAssetList.vue#L28)

File: src/components/dapp-staking-v2/my-staking/MyRewards.vue [hunk 3]
```diff
@@ -0,0 +1,97 @@
+<template>
+  <div class="wrapper--cards">
+    <div class="card">
+      <p>
+        Total Staked
```
Comment (impelcrypto) [inline src/components/dapp-staking-v2/my-staking/MyRewards.vue:5]: and here

File: src/components/dapp-staking-v2/my-staking/components/CardList.vue [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+export default defineComponent({
+  props: {
+    items: {
+      type: Array as PropType<any[]>,
+      required: true,
+    },
+  },
+  setup() {
+    const { width, screenSize } = useBreakpoints();
+    const hoverIndex = ref(-1);
```
Comment (impelcrypto) [inline src/components/dapp-staking-v2/my-staking/components/CardList.vue:43]:
```suggestion
    const hoverIndex = ref<number>(-1);
```

**Review by impelcrypto — CHANGES_REQUESTED** [review]

**Comment by impelcrypto** [comment]
@sirius651 I saw a lot of code in `files changed` are unrelated in this branch (these should be fixed in previous PRs). Perhaps you'll need to pull the latest branch of [feat/dapp-staking-v2](https://github.com/AstarNetwork/astar-apps/tree/feat/dapp-staking-v2)

**Comment by impelcrypto** [comment]
@sirius651 I've pulled latest branch of [feat/dapp-staking-v2](https://github.com/AstarNetwork/astar-apps/tree/feat/dapp-staking-v2) and fixed conflicts in [this](https://github.com/AstarNetwork/astar-apps/pull/509/commits/4385629dc43e4e493a9499f232806510c8ddb138) commit.

File: src/components/dapp-staking-v2/my-staking/MyRewards.vue (hunk 3 above)
Comment (sirius651) [reply src/components/dapp-staking-v2/my-staking/MyRewards.vue]: @impelcrypto I think all issues seems resolved as your feedback. please check again.

File: src/components/dapp-staking-v2/my-staking/MyDapps.vue [hunk 5]
```diff
@@ -5,23 +5,25 @@
...
+              <th>{{ $t('myDapps.totalEarned') }}</th>
+              <th>{{ $t('myDapps.manage') }}</th>
             </tr>
           </thead>
           <tbody>
             <tr v-for="t in items" :key="t.id">
               <td>{{ t.name }}</td>
-              <td>{{ t.stakedAmount.toLocaleString() }} ASTR</td>
-              <td>{{ t.totalEarned.toLocaleString() }} ASTR</td>
+              <td><TokenBalance :balance="t.stakedAmount" symbol="ASTR" /></td>
```
Comment (impelcrypto) [inline src/components/dapp-staking-v2/my-staking/MyDapps.vue:17]: 1. ` symbol="ASTR" `: You can get the native token symbol by `  const { nativeTokenSymbol } = useNetworkInfo();`
2. `<TokenBalance />` -> `<token-balance />`

**Review by impelcrypto — CHANGES_REQUESTED** [review]

**Review by impelcrypto — APPROVED** [review]

**Pull request merged**